BraveWallet/Crypto/Stores/SwapTokenStore.swift

```diff
@@ -392,6 +392,43 @@ public class SwapTokenStore: ObservableObject {
     }
   }
   
+  // MARK: Public 
+  
+  func prepareSwap() {
+    switch state {
+    case .error(_), .idle:
+      // will never come here
+      break
+    case .lowAllowance(let spenderAddress):
+      createERC20SwapTransaction(spenderAddress)
+    case .swap:
+      createETHSwapTransaction()
+    }
+  }
+  
+  func fetchPriceQuote(base: SwapParamsBase) {
+    guard let swapParams = swapParameters(for: base) else { return }
+    
+    updatingPriceQuote = true
+    swapController.priceQuote(swapParams) { [weak self] success, response, error in
+      guard let self = self else { return }
+      if success {
+        if let response = response {
+          self.handlePriceQuoteResponse(response, base: base)
+        }
+        self.updatingPriceQuote = false
```

Comment:
Just set this in `defer` of the closure and remove the set in the else

i.e. `defer { self.updatingPriceQuote = false }` (after `guard let self = self`)